content/docs/latest/setup/security/luks.md

```diff
@@ -0,0 +1,246 @@
+
+Example for TPM2-backed rootfs encryption with systemd-cryptenroll (Butane YAML):
+
+```yaml
+variant: flatcar
+version: 1.0.0
+storage:
+  luks:
+  - name: rootencrypted
+    wipe_volume: true
+    device: "/dev/disk/by-partlabel/ROOT"
+  filesystems:
+    - device: /dev/mapper/rootencrypted
+      format: ext4
+      label: ROOT
+systemd:
+  units:
+    - name: cryptenroll-helper.service
+      enabled: true
+      contents: |
+        [Unit]
+        ConditionFirstBoot=true
+        OnFailure=emergency.target
+        OnFailureJobMode=isolate
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        ExecStart=systemd-cryptenroll --tpm2-device=auto --unlock-key-file=/etc/luks/rootencrypted --wipe-slot=0 --tpm2-pcrs= /dev/disk/by-partlabel/ROOT
+        ExecStart=rm /etc/luks/rootencrypted
+        [Install]
+        WantedBy=multi-user.target
+```
+
+For Clevis one needs a Butane version with support for the Flatcar 1.2 Butane spec (here using a dev version built from a [PR](https://github.com/coreos/butane/pull/523)).
+Example for TPM2-backed rootfs encryption with Clevis:
+
+```yaml
+variant: flatcar
+version: 1.2.0-experimental
+storage:
+  luks:
+  - name: rootencrypted
+    wipe_volume: true
+    device: "/dev/disk/by-partlabel/ROOT"
+    clevis:
+      tpm2: true
+  filesystems:
+    - device: /dev/mapper/rootencrypted
+      format: ext4
+      label: ROOT
+```
+
+Clevis can also use a Tang server as secret store (Butane YAML):
+
+```yaml
+variant: flatcar
+version: 1.2.0-experimental
+kernel_arguments:
+  should_exist:
+    - rd.networkd=1
+storage:
+  luks:
+  - name: rootencrypted
+    device: "/dev/disk/by-partlabel/ROOT"
+    clevis:
+      tang:
+        - url: "http://tang.mycompany.com"
+          thumbprint: "Hk1VN2eKhzaVqWhXtXxXIGF5LRZt4cBWWb07I1-a0NX"
+```
+
+## A more secure setup for disk encryption
+
+When disabling auto-updates and doing one additional reboot to get rid of the Ignition kernel cmdline parameter before binding against PCRs, one can use instruct GRUB to measure the kernel and boot code when using UEFI (PCR 8+9, note that the PCR values are not set on BIOS), and tell systemd-cryptenroll to bind against more than PCR 7. There are still limitations to the achieved protection because of the mentioned lack of extension signing on the OEM partition. Other issues may also reduce the security as this is an active area of development. The GRUB PCR values will be set after first boot, but that is ok because the first boot values are special and we can't bind against them.
+
+Example for TPM2-backed rootfs encryption with systemd-cryptenroll and stronger PCR binding (requires UEFI), including disabling updates (Butane YAML):
+
+```yaml
+variant: flatcar
+version: 1.0.0
+storage:
+  files:
+    - path: /oem/grub.cfg
+      append:
+        - inline: |
+            insmod tpm
+            tpm_record_pcrs 8-9
```

Comment:
I think `tpm_record_pcrs` is not needed, but not really sure yet.

Reply:
i would remove this and make sure grub does the measuring by default. we don't want everyone to run a different configuration that we risk breaking

Reply:
I prepared a change here: https://github.com/flatcar/scripts/pull/1861